src/asset.ts

```diff
@@ -0,0 +1,152 @@
+import algosdk, { Account, Algodv2 } from 'algosdk'
+import { Config, sendGroupOfTransactions } from '.'
+import { TransactionGroupToSend, TransactionToSign } from './types/transaction'
+
+const MaxTxGroupSize = 16
+
+function* chunks<T>(arr: T[], n: number): Generator<T[], void> {
+  for (let i = 0; i < arr.length; i += n) yield arr.slice(i, i + n)
+}
+
+async function ensureAccountIsValid(client: algosdk.Algodv2, account: algosdk.Account) {
```

Comment:
Same comment as in `py` version, are these meant to be used only by optin and optout or we want to export them? If we aren't exporting them ensure you are only exposing the optIn and optOut to users